Wyam.Core/Modules/CopyFiles.cs

```diff
@@ -106,7 +106,7 @@ public IEnumerable<IDocument> Execute(IReadOnlyList<IDocument> inputs, IExecutio
                             .Select(file =>
                             {
                                 string destination = _destinationPath == null
-                                    ? Path.Combine(context.OutputFolder, PathHelper.GetRelativePath(Path.GetDirectoryName(path), Path.GetDirectoryName(file)), Path.GetFileName(file)) 
+                                    ? Path.Combine(context.OutputFolder, PathHelper.GetRelativePath(context.InputFolder, Path.GetDirectoryName(file)), Path.GetFileName(file)) 
```

Comment:
Yeah, whatever we end up with will definitely be interface based, with the interfaces going in Wyam.Common and the implementations going in Wyam.Core, similar to the rest of the abstractions. Reliance on static members is one reason I want to get off of System.IO in the first place. I envision something like a `IFileSystem` object being passing in the execution context as the entry point to file system functionality.

Gave DI a lot of thought when first architecting things. In general, I'm not a fan of canned DI frameworks (though I think the overall concept of IoC and runtime implementation selection is fine). Usually too much "magic" for my taste. You can see I like to be very explicit - heck, I'm still not a big fan of `var` :)

That said, there are two places I could see wanting to swap out different implementations for a few select interfaces: testing and the config files (for really advanced scenarios). What I may end up implementing at some point soon is a more direct factory pattern with implementation registration in the engine.